src/components/Forms/SignIn/SignIn.js

```diff
@@ -0,0 +1,215 @@
+import './SignIn.scss';
+
+export class SignIn extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      showPassword: false,
+      errors: {},
+      isFormValid: true,
+    };
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target;
+    const errors = {};
+    const [errorValue] = validate({
+      [name]: value,
+    }, name === 'password' ? passwordSchema : emailSchema);
+
+    errors[name] = errorValue || '';
+    this.setFormValidity(errors);
+    this.setState({
+      [name]: value,
+      errors,
+    });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    const { signIn, history } = this.props;
+    const {
+      email,
+      password,
+      errors,
+    } = this.state;
+    const formIsValid = this.validateForm();
+
+    if (!formIsValid) {
+      return this.setFormValidity(errors);
+    }
+    signIn({ email, password }, history);
+  }
+
+  validateForm = () => {
+    const errors = {};
+    const {
+      email,
+      password,
+    } = this.state;
+    const [emailError] = validate({ email }, emailSchema);
+    const [passwordError] = validate({ password }, passwordSchema);
+
+    errors.email = emailError || '';
+    errors.password = passwordError || '';
+    return this.setFormValidity(errors);
+  }
+
+  setFormValidity = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach((value) => {
+      if (value.length > 0) {
+        valid = false;
+      }
+    });
+    this.setState({ isFormValid: valid });
+
+    return valid;
+  }
+
+  toggleVisibility = () => {
+    this.setState(prevState => ({
+      showPassword: !prevState.showPassword,
+    }));
+  }
+
+  resetModal = () => {
+    const { close } = this.props;
+    this.setState({
+      email: '',
+      password: '',
+      showPassword: false,
+      errors: {},
+      isFormValid: true,
+    });
+    close();
+  }
+
+  render() {
+    const {
+      email,
+      password,
+      showPassword,
+      errors,
+      isFormValid,
+    } = this.state;
+    const {
+      ui: {
+        loading,
+        modalOpen,
+        modal,
+      },
+    } = this.props;
+    const loader = <Loader type="BallTriangle" color="#fff" height={18} width={79} />;
+
+    return (
+      <Modal close={this.resetModal} open={modalOpen && modal === 'signin'}>
+        <div className="form">
+          <h3 className="form-header">Sign in</h3>
+          <p className="form-text">
+            Sign in to experience author’s haven great community of creative minds.
+            Follow your favourite authors and interact with the articles you love
+          </p>
+          <form style={{ display: 'flex', flexDirection: 'column', justifyContent: 'center' }}>
+            <Input
+              name="email"
+              value={email}
+              type="email"
+              handleChange={this.handleChange}
+              placeholder="john.doe@foo.bar"
```

Comment:
Good work with the PR, it works fine on my Local. The loader on the signup page gives a nice user experience. For the signup modal, I noticed you had placeholders for both username and password. I think "Enter your email" and "Enter password" or something similar to those would be a better placeholder. Overall, nice work.

Reply:
Thanks, @Lundii. Could you be more specific about what you mean by `better placeholder`? What is your metric for judgement? Thanks.

Reply:
> Thanks, @Lundii. Could you be more specific about what you mean by `better placeholder`? What is your metric for judgement? Thanks.

Okay. I checked some sites and I noticed most of them used similar placeholders. Looking at the Input, it looks like an email and password are already typed in it. That's why I think the suggestion I gave is a better placeholder. What do you think?
<img width="315" alt="Screenshot 2019-09-02 at 6 51 17 PM" src="https://user-images.githubusercontent.com/37886814/64129415-b9b50880-cdb3-11e9-978b-2f2ef9f25ef8.png">

Reply:
While trying to come up with my LMS output for [Output 6.1: Mobile Design, User Research and, UI/UX](https://andela.instructure.com/courses/353/assignments/2704), I went through one of the linked videos there. In that video, Richard Fulcher (A Designer for Google Material Design) talked about user research and experience. And what I deduced from that is rather than try to redesign things users are used to, just make it easier for them to comprehend by designing in a way that spells out the task for them as much as possible. And thus, my choice for the placeholder I decided to use. I made the `password visibility toggle switch` darker than the placeholder text to hint the user that one is active while the other is inactive. Users are used to placeholder texts as it's been in existence for a long time and the input field's label alongside the placeholder text "in my opinion" points the user in the right direction. I also have a few instances where this concept has been used successfully. 
<img width="1129" alt="Screenshot 2019-09-02 at 21 14 00" src="https://user-images.githubusercontent.com/44305641/64132452-e246fd80-cdc7-11e9-98fc-772b3e5577e8.png">
This is [Trello's](https://trello.com/login) implementation of the login form and I think they must have done really good research to come up with this implementation.
And then, the whole team has approved this, so I suggest we move on to more pressing issues and revisit this if need be. What do you think?

Reply:
> While trying to come up with my LMS output for [Output 6.1: Mobile Design, User Research and, UI/UX](https://andela.instructure.com/courses/353/assignments/2704), I went through one of the linked videos there. In that video, Richard Fulcher (A Designer for Google Material Design) talked about user research and experience. And what I deduced from that is rather than try to redesign things users are used to, just make it easier for them to comprehend by designing in a way that spells out the task for them as much as possible. And thus, my choice for the placeholder I decided to use. I made the `password visibility toggle switch` darker than the placeholder text to hint the user that one is active while the other is inactive. Users are used to placeholder texts as it's been in existence for a long time and the input field's label alongside the placeholder text "in my opinion" points the user in the right direction. I also have a few instances where this concept has been used successfully.
> <img alt="Screenshot 2019-09-02 at 21 14 00" width="1129" src="https://user-images.githubusercontent.com/44305641/64132452-e246fd80-cdc7-11e9-98fc-772b3e5577e8.png">
> This is [Trello's](https://trello.com/login) implementation of the login form and I think they must have done really good research to come up with this implementation.
> And then, the whole team has approved this, so I suggest we move on to more pressing issues and revisit this if need be. What do you think?

Yeah. Seeing where you got the idea from and thinking about it, it would be cool to always try something new and innovative. I am okay with it now. Good work on the research.

Reply:
I think I like the placeholder
It gives the user an example on how to fill up the fields
I think every UI should be easy to use by giving examples on their pages
Nice work everyone